base/REPLCompletions.jl

```diff
@@ -141,6 +141,36 @@ function complete_path(path::AbstractString, pos)
             push!(matches, id ? file * (@windows? "\\\\" : "/") : file)
         end
     end
+
+    if use_envpath && length(dir) == 0
+        # Look for files in PATH as well
+        local pathdirs = split(ENV["PATH"], @unix? ":" : ";")
+
+        for pathdir in pathdirs
+            local actualpath
+            try
+                actualpath = realpath(pathdir)
+            catch
+                # Bash doesn't expect every folder in PATH to exist, so neither shall we
+                continue
+            end
+
+            if actualpath != pathdir && in(actualpath,pathdirs)
+                # Remove paths which (after resolving links) are in the env path twice.
+                # Many distros eg. point /bin to /usr/bin but have both in the env path.
+                continue
+            end
+
+            local filesinpath = readdir(pathdir)
+
+            for file in filesinpath
+                if startswith(file, prefix) && isexecutable(joinpath(pathdir, file))
```

Comment:
Not sure about that, there might be non executable files in PATH. In a perfect world, we would also filter based on whether the current user is able to execute the file in question (like bash does), but that I don't see a nice way to do that (latest bash uses glibc's `eaccess` or `access` if available, otherwise it tries to figure on its own).
Implementing something like that might be something we want to do in a follow up.